stackv4/unikernel.ml

```diff
@@ -13,33 +13,38 @@ module Main (C:CONSOLE) (S:STACKV4) = struct
 
   let start console s =
 
-    C.log_s console
-      (sprintf "IP address: %s\n"
-        (String.concat ", " (List.map Ipaddr.V4.to_string (S.IPV4.get_ip (S.ipv4 s)))))
-    >>= fun () ->
+    let ips = List.map Ipaddr.V4.to_string (S.IPV4.get_ip (S.ipv4 s)) in
+    C.log_s console (sprintf "IP address: %s\n" (String.concat ", " ips))
 
-    S.listen_udpv4 s 53 (
+    >>= fun () ->
+    let loc_port = 53 in
+    S.listen_udpv4 s loc_port (
       fun ~src ~dst ~src_port buf ->
-        C.log_s console "got udp on 53"
+        C.log_s console
+          (red "UDP %s.%d > %s.%d: \"%s\""
+             (Ipaddr.V4.to_string src) src_port
+             (Ipaddr.V4.to_string dst) loc_port
+             (Cstruct.to_string buf))
     );
 
-    S.listen_tcpv4 s 8080 (
+    let loc_port = 8080 in
+    S.listen_tcpv4 s loc_port (
       fun flow ->
-        let dst, dst_port = T.get_dest flow in
+        let rem, rem_port = T.get_dest flow in
```

Comment:
You should probably spell this one out then, because it's certainly not a remainder. :)